torchvision/prototype/datasets/utils/_internal.py

```diff
@@ -331,3 +331,7 @@ def read_flo(file: BinaryIO) -> torch.Tensor:
     width, height = fromfile(file, dtype=torch.int32, byte_order="little", count=2)
     flow = fromfile(file, dtype=torch.float32, byte_order="little", count=height * width * 2)
     return flow.reshape((height, width, 2)).permute((2, 0, 1))
+
+
+def hint_sharding(datapipe: IterDataPipe[D]) -> IterDataPipe[D]:
+    return ShardingFilter(datapipe)
```

Comment:
Because by default the `ShardingFilter` is a no-op, but this isn't clear from just using the class. By wrapping it in a `hint_sharding` function, I hope it is more clear that this is the point where sharding will happen if it is enabled, but nothing will happen by default.

Reply:
I see. Would functional API would help? `dp = dp.sharding_filter()` I am totally fine with your `hint_sharding`. If `sharding_filter` couldn't provide such hint about no-op, we may need to change the name of this API IMO.

Reply:
That maybe feasible here, but I don't think we can say the same about `Shuffler`, because that won't be a no-op by default. So an implementation would look something like

```python
def hint_shuffling(dp):
    return Shuffler(dp, enabled=False)
```

Since we need a `hint` function anyway, I think we should keep it consistent and also have one for sharding.